### ISSUE TITLE
Suportar versões atuais de Ubuntu, Debian, OpenWRT

### ISSUE THREAD

**Issue by hmh** [simetnicbr/simet-ma#8]
Status:

- [ ] Ubuntu 26.04 LTS (Resolute Raccoon) -- 64bit only, not validated yet for uutils/coreutils
- [x] Ubuntu 24.04 LTS (Noble Numbat) -- 64bit only
- [x] Ubuntu 22.04 LTS (Jammy Jellyfish) -- 64bit only
- [x] Ubuntu 20.04 LTS (Focal Fossa) -- 64bit only
- [x] Debian 13 (Trixie) -- 64bit only
- [x] Debian 12 (Bookworm) -- x86: 32 and 64-bit;  ARM: 64-bit only
- [x] Debian 11 (Bullseye) -- x86: 32 and 64-bit;  ARM: 64-bit only
- [x] Debian 10 (Buster) -- x86: 32 and 64-bit;  ARM: 64-bit only -- limited support, EOL soon
- [x] LinuxMint 18.x, 19.x, 20.x, 21.x, 22.x
- [x] OpenWRT 18.06.x ar71xx
- [x] OpenWRT 19.07 ar71xx, ath79
- [x] OpenWRT 21.02 ath79
- [x] TurrisOS 6.4+ (Omnia) -- must install package "simetbox-openwrt-turrisos"
- [x] OpenWRT 22.x ath79, mt7621
- [x] OpenWRT 23.x mt7621, filogic
- [x] OpenWRT 24.x mt7621, filogic
- [x] RPI Raspbian -- apenas via código fonte (32-bit). Note que RPI 64-bit é atendida via Debian ARM64.

Em geral, outros sistemas operacionais e versões funcionam, mas com alguns problemas de integração e/ou empacotamento.

ARM 32-bits funciona perfeitamente, porém não distribuímos binários.  Basta compilar diretamente do código fonte.

**Comment by farribeiro** [comment]
Olá, @hmh!

Como ficou o pacote do agente CLI do simet?

Como disse anteriormente, na issue #6 é extremamente vantajoso a inclusão deste simet ao agente ISPTools.

Abraços
Fábio Rodrigues Ribeiro

**Comment by hmh** [comment]
O agente CLI está empacotado faz tempo, mas só Ubuntu/Debian por enquanto.  Tenho feito inclusive "pré-releases" do dito cujo já compilado via github.

Um exemplo em docker também faz parte do pacote faz tempo.

Edit: o agente CLI não está em beta há algum tempo.  Visualizador está disponível, ele roda o navegador para acessar os resultados.

**Comment by farribeiro** [comment]
E a adesão, tem telemetria?

**Comment by hmh** [comment]
Edit: na versão de produção temos telemetria de:

* qual a versão do medidor (para detectarmos problemsa de atualização na frota, etc)
* quando um medidor registrou-se pela primeira vez, e data de último contato do medidor (para sabermos se é um medidor ativo ou não e podermos limpar registros antigos)
* qual o nome (Ubuntu, Debian, Mint...) e versão do sistema operacional em que o medidor foi instalado (para direcionar esforços de suporte e manutenção de compatibilidade)

O usuário pode, opcionalmente, configurar o medidor para enviar alguns metadados para seu registro (um hostname -- não precisa ser o do computador, e uma descrição).  Usado por ISPs e organizações que possuem muitos medidores.

Fora isso, nas *medições*, guardamos as versões do medidor e sistema operacional, além da geolocalização (com precisão reduzida para "setor censitário" -- não guardamos a geolocalização exata).

**Comment by pedroaugustosmribeiro** [comment]
Poderia preparar um pacote para o Raspberrypi 1B (Raspbian/Debian Buster 10.x armhf), por favor?

**Comment by hmh** [comment]
O pacote fonte (debian) funciona no Rasbpian, basta baixar ele no SDK, e fazer o build.   Mas a RPI 1B tem pouca capacidade de CPU e conectividade, não sabemos o quanto vai medir.

Em RPI 4B, mede perto de 1Gbps.  Em RPI 3, é limitado pelo bus interno da RPI e pode ficar errático quando lota esse bus.

Colocar os pacotes fonte em um formato mais fácil de usar para RPI (junto com um guia, se necessário) está na lista de tarefas, mas não temos uma previsão neste momento.

Edit: os binários de instalação e nosso repositório APT suporta ARM64 (em Debian e Ubuntu).   Infelizmente, ARM 32-bits só é suportado via build direto do código fonte, devido a um bug do QEMU.

**Comment by pedroaugustosmribeiro** [comment]
Ok, obrigado. Já consegui compilar a partir do próprio repositório e rodar no RPI 1B. Não serve para testes de velocidade (a interface dele é limitada a 100 Mbps), mas pode servir para verificar conectividade, latência, jitter, etc.

**Comment by hmh** [comment]
@pedroaugustosmribeiro bom, ele deveria servir para medições até perto de 100Mbps, se não "lotar" a CPU da RPI 1B.

**Comment by pedroaugustosmribeiro** [comment]
sim, porém minha conexão é de 300 Mbps, vai saturar de todo modo. Uso a RPI 1B porque é o que eu tenho, modelo 4 serial o ideal mesmo.

**Comment by hmh** [comment]
Atualizei diversos comentários para refletir a atualidade na versão de produção.

Mais detalhes sobre telemetria e dados que coletamos encontram-se na política de privacidade do medidor. Seguimos a lei brasileira (LGPD).

**Comment by slipttees** [comment]
Ubuntu 26.04 LTS?

**Comment by hmh** [comment]
É necessário validar todas as funções do GNU coreutils utilizadas no motor de medição em Ubuntu 26, já que por padrão eles usam uma reescrita em rust do coreutils que não implementa de forma completa a funcionalidade do GNU coreutils.

Funcionar vai, mas não temos a mínima ideia se vai funcionar corretamente em 100% dos casos sem essa validação.

Você pode baixar o código fonte, compilar com dpkg-buildpackage, e instalar os pacotes deb resultantes para testar, se quiser. Tenha **absoluta** certeza de que atualizou o Ubuntu 26 antes de tentar compilar ou instalar. Principalmente se usar imagens em container, etc.

**Comment by farribeiro** [comment]
Poderia por gentileza atualizar os dockerfiles? E tb se tiver imagens no hub

**Comment by slipttees** [comment]
[build-simet-lmapd-ubuntu26.04.txt](https://github.com/user-attachments/files/28016467/build-simet-lmapd-ubuntu26.04.txt)
[build-simet-ma-ubuntu26.04.txt](https://github.com/user-attachments/files/28016466/build-simet-ma-ubuntu26.04.txt)

> É necessário validar todas as funções do GNU coreutils utilizadas no motor de medição em Ubuntu 26, já que por padrão eles usam uma reescrita em rust do coreutils que não implementa de forma completa a funcionalidade do GNU coreutils.
> 
> Funcionar vai, mas não temos a mínima ideia se vai funcionar corretamente em 100% dos casos sem essa validação.
> 
> Você pode baixar o código fonte, compilar com dpkg-buildpackage, e instalar os pacotes deb resultantes para testar, se quiser. Tenha **absoluta** certeza de que atualizou o Ubuntu 26 antes de tentar compilar ou instalar. Principalmente se usar imagens em container, etc.

precisei alterar o CMakeLists.txt no simet-lmapd

```
- cmake_minimum_required(VERSION 3.0.0 FATAL_ERROR) 
+ cmake_minimum_required(VERSION 3.0...4.2 FATAL_ERROR) 
```
e no CMakeLists.txt do simet-ma

```
- cmake_minimum_required(VERSION 2.6)
+ cmake_minimum_required(VERSION 2.6...4.2)
```

Para contornar os binário estáticos 
```
- set(CMAKE_FIND_LIBRARY_SUFFIXES ".a")
- set(BUILD_SHARED_LIBS OFF)
- set(CMAKE_EXE_LINKER_FLAGS "-static")

+ option(STATIC_BUILD "Build fully static binaries" OFF)

+ if(STATIC_BUILD)
+    set(CMAKE_FIND_LIBRARY_SUFFIXES ".a")
+    set(BUILD_SHARED_LIBS OFF)
+    set(CMAKE_EXE_LINKER_FLAGS "-static")
+ endif()
```

e precisei contornar o Base64.c linha 210
 
```
 - char *pos;
 + const char *pos; 
```

o lemon.c linhas 613 a 618
```
- void FindRulePrecedences();
- void FindFirstSets();
- void FindStates();
- void FindLinks();
- void FindFollowSets();
- void FindActions();

+ void FindRulePrecedences(struct lemon *);
+ void FindFirstSets(struct lemon *);
+ void FindStates(struct lemon *);
+ void FindLinks(struct lemon *);
+ void FindFollowSets(struct lemon *);
+ void FindActions(struct lemon *);
```

após essas alterações compilou como pode ver nos logs. Agora é testar.

**Comment by hmh** [comment]
Versão 0.18.14.2 suporta gcc 15, cmake 4.2 (Ubuntu 26 com GNU coreutils).  Vai precisar também do simet-lmapd 0.15.7.  Não foi ainda validado para funcionar em Ubuntu 26 com o uutils/coreutils, que é o "coreutils padrão" do Ubuntu 26.

**Comment by slipttees** [comment]
Está a funcionar no 26.04. Posso enviar logs e medições para avaliações se precisar!
